framework/test/mocha/unit/components/storage/entities/account.js

```diff
@@ -628,6 +630,112 @@ describe('Account', () => {
 		});
 
 		describe('filters', () => {
+			describe('asset_contains', () => {
+				let aliceAccount = null;
+				let bobAccount = null;
+
+				beforeEach(async () => {
+					aliceAccount = new accountFixtures.Account({
+						asset: {
+							name: 'Alice',
+							company: 'Acme Corp',
+						},
+					});
+					bobAccount = new accountFixtures.Account({
+						asset: {
+							name: 'Bob',
+							company: 'Acme Corp',
+							tags: ['blockchain', 'crypto'],
+						},
+					});
+
+					await AccountEntity.create([aliceAccount, bobAccount]);
+				});
+
+				it('should return single result when asset value contains unique property at the top level', async () => {
+					const filters = { asset_contains: { name: 'Alice' } };
+					const accounts = await AccountEntity.get(filters);
+
+					expect(accounts.length).to.be.eql(1);
+					expect(accounts[0].address).to.be.eql(aliceAccount.address);
```

Comment:
I tried to match `expect(accounts[0]).to.be.eql(aliceAccount);` but it seems that the query returns some more fields.

Reply:
That's is because the fixture doesn't match the storage. As we are testing the filter and not the create method, I reckon it's fine to assert the address only.